src/SkyDrop.Core/Services/StorageService.cs

```diff
@@ -51,8 +51,16 @@ public void DeleteSkyFile(SkyFile skyFile)
 
         public UploadAverage GetAverageUploadRate()
         {
-            var uploadAverage = realm.All<UploadAverage>().FirstOrDefault();
-            return uploadAverage ?? new UploadAverage();
+            try
+            {
+                var uploadAverage = realm.All<UploadAverage>().First();
+                return uploadAverage;
+            }
+            catch (InvalidOperationException)
+            {
+                log.Trace("No uploaded file in UploadAverage table");
+                return new UploadAverage();
+            }
         }
```

Comment:
Rider kept breaking here when debugging, idk why, but changing it to this resolved it and behaviour is equivalent